core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -68,7 +68,7 @@ class KafkaMetadataLog(
 
     val appendInfo = log.appendAsLeader(records.asInstanceOf[MemoryRecords],
       leaderEpoch = epoch,
-      origin = AppendOrigin.Coordinator)
+      origin = AppendOrigin.Replication)
```

Comment:
I intended to add a new `AppendOrigin` called `Leader` for this case, but then think it implies the same as the Replication, so just reuses it.

Reply:
I was going to ask about adding a new `AppendOrigin`. I agree that the behavior should be the same as for `Replication`, but it seems like it could lead to confusion. Maybe we could add an `AppendOrigin.RaftLeader`. Then we can add a simple comment which emphasizes that the Raft leader is responsible for assigning offsets. That would also allow us to revert the changes in `LogCleanerTest`.

Reply:
Added a `AppendOrigin.RaftLeader` and also reverted the changes in `LogCleanerTest `